### PR TITLE
Moving image.csv processing to an earlier step

### LINKED ISSUE #35: Splitting full site annotation in 4.image-and-segmentation-qc
This code block is giving me some trouble:

https://github.com/broadinstitute/pooled-cell-painting-profiling-recipe/blob/068c7eae79f56a50732fdf173902dc596c95ce3f/0.preprocess-sites/8.image-and-segmentation-qc.py#L202-L204

A couple notes:
* Is the site_full always going to be coded in this way? i.e. `Plate`, then `Well`, then `Site`?
* I've already encountered when site_full is not split by `-` (in a separate experiment, we delimit by underscore)

So, this is a very fragile way of handling this split. A couple of solutions:

* We **always** output site_full with the same convention - this is still pretty fragile, since most scientists will forget the standard)
* We somehow propagate `plate`, `well`, and `site number` information earlier and do not rely on this split at all. Is this info in the `Image.csv` file in a seprate metadata column already?

@ErinWeisbart - we should either resolve this or drop this before version 0.1

### LINKED ISSUE #38: Bugs in 4.image-and-segmentation-qc.py
Noting two things to be fixed in a future pull request:

### `force` not defined

```python
(pooled-cp) wm962-fdf:0.preprocess-sites gway$ python 4.image-and-segmentation-qc.py
Traceback (most recent call last):
  File "4.image-and-segmentation-qc.py", line 113, in <module>
    if check_if_write(output_file, force, throw_warning=True):
NameError: name 'force' is not defined
```

### Error concatenating

```python
Done concatenating image files
Traceback (most recent call last):
  File "4.image-and-segmentation-qc.py", line 478, in <module>
    image_corr_df = pd.concat(image_corr_list).drop_duplicates(subset="site").reset_index()
  File "/Users/gway/miniconda3/envs/pooled-cp/lib/python3.7/site-packages/pandas/core/reshape/concat.py", line 281, in concat
    sort=sort,
  File "/Users/gway/miniconda3/envs/pooled-cp/lib/python3.7/site-packages/pandas/core/reshape/concat.py", line 329, in __init__
    raise ValueError("No objects to concatenate")
ValueError: No objects to concatenate
```

These should be quick fixes

### PR DESCRIPTION
**Pull request by gwaygenomics**
closes #35
closes #38

The commit messages contain the story of the PR.

In addition to closing the two issues above, I also add two additional summary figures. Since they contain data, I will slack them to you. The two figures are:

* Number of unique genes/guides per site
* Perturbation coverage

Calvin has asked for these for like a month now! Finally getting around to it :)

### CODE REVIEW

**Comment by gwaygenomics** [comment]
Also note #40 in which I explain why the last couple of lines in `4.image-and-segmentation-qc.py` needed to be removed.

File: 0.preprocess-sites/4.image-and-segmentation-qc.py [hunk 1]
```diff
@@ -360,32 +350,36 @@
 # x = std dev of intensity (to find images that have unusually bright spots)
 # y = % image that is saturated (to find images that are unusually bright)
 # Look at points off cluster where x > 1
```
Comment (ErinWeisbart) [inline 0.preprocess-sites/4.image-and-segmentation-qc.py:352]:
```suggestion
# Look at points off cluster where y > 1
```
oops. my fault this typo is here in the first place.

File: 0.preprocess-sites/4.image-and-segmentation-qc.py [hunk 2]
```diff
@@ -407,7 +401,7 @@
 # x = std dev of intensity (to find images that have unusually bright spots)
 # y = % image that is saturated (to find images that are unusually bright)
 # Look at points off cluster where x > .2
```
Comment (ErinWeisbart) [inline 0.preprocess-sites/4.image-and-segmentation-qc.py:403]:
```suggestion
# Look at points off cluster where y > .2
```

**Comment by gwaygenomics** [comment]
The code has changed enough since last review that it is probably worth another quick look (when you get a chance) before merging @ErinWeisbart

**Comment by ErinWeisbart** [comment]
I'm a little overwhelmed at trying to read it all at once, but if I step through the commits it makes sense, so I say:

**Pull request merged**